### PR TITLE
Update code coverage orbs

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Update the [CircleCI orbs](https://circleci.com/developer/orbs) for reporting code coverage to a more recent version.

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -124,6 +124,9 @@ jobs:
             - /root/.cache/pip
             - .mypy_cache
             - .pytest_cache
+      - run:
+          name: Install gpg (required by codecov orb)
+          command: apt update && apt install -y gpg
       - codecov/upload:
           when: always
           file: 'test-reports/coverage.xml'
```
Comment (valeriupredoi) [inline .circleci/config.yml:132]: what's the report.xml created above used for then?

**Comment by valeriupredoi** [comment]
not really liking this new stuff though - it says there are 11 scripts with some indirect chamges to coverage but when I try clicking on one it tells me some error occured (that might be if Codecov server is down) - but those scripts used to be listed in the table in the codecov comment on the PR page - will those be gone now and only visible at the Codecov page? See [this](https://app.codecov.io/gh/ESMValGroup/ESMValCore/pull/2206/blob/esmvalcore/preprocessor/_io.py) for one of those scripts I emntion

File: .circleci/config.yml (hunk 1 above)
Comment (bouweandela) [reply .circleci/config.yml:132]: It's used to generate the test summary page: https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/9756/workflows/2169fe35-d00d-448d-826f-eeaa47109d51/jobs/42339/tests

That looks quite boring for this pr, but if there are some failing tests or gives you a nice quick overview without having to scroll through lots of terminal output.

Comment (valeriupredoi) [reply .circleci/config.yml:132]: aha! Nice one then, thanks

**Review by valeriupredoi — APPROVED** [review]
OK progress I guess but I liked that table with up/down coverage per file, in the bot's comment

**Comment by bouweandela** [comment]
> OK progress I guess but I liked that table with up/down coverage per file, in the bot's comment

The table is only shown when there are does with changed coverage in the PR, that's why you don't see it here.

**Comment by valeriupredoi** [comment]
> > OK progress I guess but I liked that table with up/down coverage per file, in the bot's comment
> 
> The table is only shown when there are files with changed coverage in the PR, that's why you don't see it here.

I suspected that but I was a bit taken aback by those "files with indirect changes" links I saw earlier - at any rate, am merging this, we spent too much time discussing it already :grin:

**Pull request merged**